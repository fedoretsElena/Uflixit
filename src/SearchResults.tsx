import React, { Component } from 'react';

class SearchResults extends Component {

    render() {
        console.log('Props', this.props);

        return (
            <section>tesst</section>
        )
    }
}

export default SearchResults;