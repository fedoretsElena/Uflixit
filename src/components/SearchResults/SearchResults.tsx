import React, { Component } from 'react';
import { UnregisterCallback } from 'history';

import Title from '../shared/Title';
import Loader from '../shared/Loader';
import ErrorMsg from '../shared/ErrorMsg';
import MediaList from '../shared/MediaList';
import BaseMedia from '../../models/BaseMedia';
import { MediaType } from '../../models/MediaFactory';
import MediaService from '../../services/mediaService';

interface IProps {
    location: any,
    history: any
}

interface IStore {
    media: any;
    loading: boolean;
    loaded: boolean;
    error: string | null;
}

class SearchResults extends Component<IProps, IStore> {
    stopListenRoute: UnregisterCallback = () => {};
    mediaService: MediaService = new MediaService();
    state: IStore = {
        loading: false,
        loaded: false,
        error: null,
        media: []
    };

    render() {
        const { media, loaded, loading, error } = this.state;

        return (
            <>
                <Title title='Search results'
                       length={media.length}
                />
                {loading && !error && <Loader/>}
                {loaded && <MediaList items={media}
                                      type={MediaType.TVShow} />}
                {error && <ErrorMsg msg={error}/>}
            </>
        )
    }

    componentDidMount(): void {
        console.log('searching!');

        this.search();
        this.startListenRouteChanges();
    }

    componentWillUnmount(): void {
        console.log('stop!');
        this.stopListenRoute();
    }

    private search(newLocation?: any): void {
        const location: any = newLocation || this.props.location;
        const params = new URLSearchParams(location.search);

        this.setState({
            media: [],
            loading: true
        });

        this.mediaService.search(params)
            .then((res: BaseMedia[]) => {
                this.setState({
                        media: res,
                        loaded: true,
                        loading: false,
                        error: null
                    })
                }
            )
            .catch(() => this.setState({
                    error: 'There is no game according to the request.',
                    loading: false
                })
            );
    }

    private startListenRouteChanges(): void {
        this.stopListenRoute = this.props.history.listen((location: any) => this.search(location));
    }
}

export default SearchResults;