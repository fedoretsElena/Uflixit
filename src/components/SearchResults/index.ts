import SearchForm from '../Header/SearchForm';

export default SearchForm;