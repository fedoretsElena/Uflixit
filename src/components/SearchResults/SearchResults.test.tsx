import SearchResults from "./SearchResults";

jest.mock("../../data/services/mediaService");

describe("MediaDetail component", () => {
    let mockProps: { location: any; history: any };
    let component: any;

    beforeEach(() => {
        mockProps = {
            location: {
                search: "Games"
            },
            history: {
                listen: () => "Games"
            }
        };

        component = shallow(
            <SearchResults
                history={mockProps.history}
                location={mockProps.location}
            />
        );
    });

    it("should render correctly and take data", done => {
        setTimeout(() => {
            component.update();

            expect(component.instance().state.media).toHaveLength(1);
            done();
        });
    });

    it("cancelSearchRequest should initialized after calling search func", () => {
        component.instance().search();

        setTimeout(() => {
            expect(component.instance().cancelSearchRequest).toBeDefined();
        });
    });
});
