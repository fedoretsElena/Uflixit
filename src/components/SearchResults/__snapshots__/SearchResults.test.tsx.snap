// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaDetail component should render correctly and take data 1`] = `
<Fragment>
  <Title
    length={0}
    title="Search results"
  />
  <Loader />
</Fragment>
`;
